test/helpers.spec.ts

```diff
@@ -47,6 +47,18 @@ describe('helpers', () => {
         .it('some string', 'Hello !#{3}test[1-5]');
     });
 
+    t.describe('fromRegExp', (t) => {
+      t.it('with RegExp object', /[A-D0-9]{4}-[A-D0-9]{4}/)
+        .it(
+          'with negated RegExp object with case insensitive flag',
+          /[^a-t0-9]{4}/i
+        )
+        .it(
+          'with RegExp object and case insensitive flag',
+          /[A-D0-9]{4}-[A-D0-9]{4}/i
+        );
```

Comment:
```suggestion
      t.it('with static string', 'Hello World!')
        .it('with static RegExp', /Hello World!/)
        .it('with dynamic string', '[A-D0-9]-[A-D0-9]-A')
        .it('with dynamic RegExp', /[A-D0-9]-[A-D0-9]-A/)
        .it('with wildcard character', /./)
        .it('with optional character', /A?-[A-D0-9]?-[a-d0-4]?/)
        .it('with optional repetition', /A*-[A-D0-9]*-[a-d0-4]*/)
        .it('with required repetition', /A+-[A-D0-9]+-[a-d0-4]+/)
        .it('with quantifier', /A{2}-[A-D0-9]{4}-[a-d0-4]{6}/)
        .it('with quantifier ranges', /A{2,6}-[A-D0-9]{4,6}-[a-d0-4]{6,8}/)
        .it('with case insensitive flag', /[A-D0-9]{10}/i)
        .it('with negation and case insensitive flag', /[^a-t0-7]{10}/i)
        .it('with negation', /[^A-Za-y0-9]{10}/);
```

Could you add support for the missing cases as well?

Reply:
I could, it will take some time and it will make the function larger and more complex than it currently is which I assume is a concern from previous reviews.

Reply:
@faker-js/maintainers @faker-js/members `@anyone`: Any preferences?

Reply:
I wouldn't actually put any restrictions on the repetition part. I think something like `while (faker.datatype.boolean()) { [...] }` instead. But a static number that gets generated upfront might be easier to handle.

For the wildcard, all functions from the string module (including special characters) can be used IMO.

Reply:
We could use an option to overwrite what `.` is and it can be used as a base for `[^something]`.

Reply:
> `while (faker.datatype.boolean()) { [...] }`

Wont this make it so that higher repetitions are rarer?

> But a static number that gets generated upfront might be easier to handle.

I agree but the concern with that is like what I stated above, what should be the maximum number of repetitions? If it randomly generates some large number, it will still work but its kinda unreasonable, no?

> For the wildcard, all functions from the string module (including special characters) can be used IMO.

I see a function `faker.string.sample()`, is this function suitable for generating wildcard characters?
There is also another concern, supposedly `.` matches all characters except `\n`. Chinese, Cyrillic etc characters also need to be included but to include those is also a huge pain, and I have no idea where to start with those. Can I sweep this concern under the rug?

> We could use an option to overwrite what `.` is and it can be used as a base for `[^something]`.

I think this is not appropriate. If the user wants to give their own definition, they can just do it in normal character sets, i.e `/[a-z0-9]/`.

Reply:
> Wont this make it so that higher repetitions are rarer?

Yes, you are right. I wanted to include the current interaction count as a `probability` parameter in the function (but forgot to actually write it). This would then look something like this: `while (faker.datatype.boolean(1 / currentIteration)) { [...] }` which equates to a logarithmic distribution of repetitions.

> I agree but the concern with that is like what I stated above, what should be the maximum number of repetitions? If it randomly generates some large number, it will still work but its kinda unreasonable, no?

If this is way easyer for you to implement for you - go for it and hard code the max number to an arbitrary number (like 25). Don't think to much about it. Altho, one could argue that since it is a wildcard, an infinite number of repetitions is expected by design.

> I see a function faker.string.sample(), is this function suitable for generating wildcard characters?
There is also another concern, supposedly . matches all characters except \n. Chinese, Cyrillic etc characters also need to be included but to include those is also a huge pain, and I have no idea where to start with those. Can I sweep this concern under the rug?

Yes, this is perfect. Altho `.` matches anything I would argue that it is simple enough to return any character from our string methods. Keep in mind that this should be a helper function, not a full blown library on its own^^

> I think this is not appropriate. If the user wants to give their own definition, they can just do it in normal character sets, i.e /[a-z0-9]/.

I'm with you. RegExp lets you do some fantastic things. If you want a specific char set, don't use a wildcard matcher.